### PR TITLE
Code scanning cleanups

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt [hunk 1]
```diff
@@ -148,7 +148,7 @@ private fun mapLicense(
         //       which cannot be parsed to an SpdxExpression. A better solution could be to automatically
         //       convert the strings into a form that can be parsed, then the mapping could be applied globally.
         LicenseFinding(license.mapLicense(detectedLicenseMapping), location)
-    }.onSuccess { licenseFinding ->
+    }.map { licenseFinding ->
```
Comment (sschuberth) [inline plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt:151]: @nnobelis please double-check.

File: plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt
Comment (mnonnenmacher) [inline plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt:1]: Commit message: Typo in "onSuccess". But the message also confused me at first, it should say something like "`onSuccess` returns the original result, so the copied license finding is never used".

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Review by mnonnenmacher — APPROVED** [review]

File: plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt (hunk 1 above)
Comment (nnobelis) [reply plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt:151]: We discussed with @mnonnenmacher : it looks wrong and should have been picked up by the `FossIdLicenseMappingTest`. I will look at it in the future.

Comment (nnobelis) [reply plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt:151]: I found why the test was previously passing. Please see https://github.com/oss-review-toolkit/ort/pull/8495.

**Pull request merged**